btf/btf_test.go

```diff
@@ -324,8 +324,8 @@ func TestLoadSpecFromElf(t *testing.T) {
 		}
 
 		t.Run("Handle", func(t *testing.T) {
+			testutils.SkipIfNotSupported(t, haveMapBTF())
```

Comment:
Nit: This change is not required anymore?

Reply:
It is. `loader-el.elf` contains a VAR, the BTF blob is loaded in full, and `NewHandle()` only returns `ErrNotSupported` on kernels <4.18. On 4.19:

            btf_test.go:330: Can't load BTF: invalid argument: [12] Invalid kind:14 (25 line(s) omitted)

I think we could eventually call `haveBTF`, `haveMapBTF` and `haveProgBTF` (in that order) in `NewHandle()`.. Then we could accurately wrap an error with the corresponding missing BTF support level. Maybe good for a follow-up?

Reply:
Yep, maybe an alternative could be adding a probe method on each BTF kind. If loading a BTF blob fails, we look for the highest kind in the spec and call its probe method to check if the kernel supports it. (the goal being accurate ErrNotSupported from NewHandle)